Issue with PySide6>=6.2.2

There are lots of errors in the CI tests with `PySide6>=6.2.2` similar to this one:
```
____________________________ test_ginput[scatter] _____________________________
AttributeError: can't set attribute
The above exception was the direct cause of the following exception:
sx = <module 'silx.sx' from 'C:\\projects\\silx\\venv_test\\lib\\site-packages\\silx\\sx\\__init__.py'>
qapp = <PySide6.QtWidgets.QApplication(0x2743e6937b0) at 0x0000027446667088>
qapp_utils = <silx.gui.utils.testutils.TestCaseQt testMethod=runTest>
plot_kind = 'scatter'
    @pytest.mark.parametrize("plot_kind", ["plot", "imshow", "scatter"])
    def test_ginput(sx, qapp, qapp_utils, plot_kind):
        """Test ginput function
    
        This does NOT perform interactive tests
        """
        create_plot = getattr(sx, plot_kind)
>       plt = create_plot()
venv_test\lib\site-packages\silx\test\test_sx.py:180: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
venv_test\lib\site-packages\silx\sx\_plot.py:348: in scatter
    plt = ScatterView()
venv_test\lib\site-packages\silx\gui\plot\ScatterView.py:79: in __init__
    plot = PlotWidget(parent=self, backend=backend)
venv_test\lib\site-packages\silx\gui\plot\PlotWidget.py:422: in __init__
    self._yRightAxis = items.YRightAxis(self, self._yAxis)
venv_test\lib\site-packages\silx\gui\plot\items\axis.py:481: in __init__
    Axis.__init__(self, plot)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <silx.gui.plot.items.axis.YRightAxis(0x2744c27b970) at 0x000002744D428A88>
plot = <silx.gui.plot.PlotWidget.PlotWidget(0x27441dcd0e0) at 0x000002744D7A3148>
    def __init__(self, plot):
        """Constructor
    
        :param silx.gui.plot.PlotWidget.PlotWidget plot: Parent plot of this
            axis
        """
>       qt.QObject.__init__(self, parent=plot)
E       SystemError: <slot wrapper '__init__' of 'PySide6.QtCore.QObject' objects> returned a result with an error set
```

--- a/ci/requirements-pinned.txt
+++ b/ci/requirements-pinned.txt
@@ -8,6 +8,3 @@ pybind11  # Required to build pyopencl
 # downloaded from https://www.lfd.uci.edu/~gohlke/pythonlibs/#pyopencl
 # Anyway, we don't test OpenCL on appveyor
 pyopencl == 2020.3.1; sys_platform == 'win32'
-
-# Until PySide6>=6.2.2 passed CI
-PySide6 == 6.2.1

--- a/src/silx/gui/plot/items/axis.py
+++ b/src/silx/gui/plot/items/axis.py
@@ -1,7 +1,7 @@
 # coding: utf-8
 # /*##########################################################################
 #
-# Copyright (c) 2017-2021 European Synchrotron Radiation Facility
+# Copyright (c) 2017-2022 European Synchrotron Radiation Facility
 #
 # Permission is hereby granted, free of charge, to any person obtaining a copy
 # of this software and associated documentation files (the "Software"), to deal
@@ -480,26 +480,10 @@ class YRightAxis(Axis):
         """
         Axis.__init__(self, plot)
         self.__mainAxis = mainAxis
-
-    @property
-    def sigInvertedChanged(self):
-        """Signal emitted when axis orientation has changed"""
-        return self.__mainAxis.sigInvertedChanged
-
-    @property
-    def sigScaleChanged(self):
-        """Signal emitted when axis scale has changed"""
-        return self.__mainAxis.sigScaleChanged
-
-    @property
-    def _sigLogarithmicChanged(self):
-        """Signal emitted when axis scale has changed to or from logarithmic"""
-        return self.__mainAxis._sigLogarithmicChanged
-
-    @property
-    def sigAutoScaleChanged(self):
-        """Signal emitted when axis autoscale has changed"""
-        return self.__mainAxis.sigAutoScaleChanged
+        self.__mainAxis.sigInvertedChanged.connect(self.sigInvertedChanged.emit)
+        self.__mainAxis.sigScaleChanged.connect(self.sigScaleChanged.emit)
+        self.__mainAxis._sigLogarithmicChanged.connect(self._sigLogarithmicChanged.emit)
+        self.__mainAxis.sigAutoScaleChanged.connect(self.sigAutoScaleChanged.emit)
 
     def _internalSetCurrentLabel(self, label):
         self._getBackend().setGraphYLabel(label, axis='right')

